### PR TITLE
bugfix remove_extra_files when only bower.json present in root

### PR DESCRIPTION
**Pull request by gingray**
https://github.com/rharriso/bower-rails/issues/175 this pull request try to solve this issue. Also i've add tests seems that branch with only bower.json file (without Bowerfile) haven't been tested well.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -5,4 +5,11 @@
 $:.unshift File.expand_path('../../lib', __FILE__)
 
 require 'rspec'
-require 'bower-rails'
\ No newline at end of file
+require 'bower-rails'
+
+class Hash
```
Comment (SergeyKishenin) [inline spec/spec_helper.rb]: Seems like this change is unused

**Comment by SergeyKishenin** [comment]
Also, please update Changelog.

**Comment by gingray** [comment]
@SergeyKishenin you mean increment version to v0.10.1 and put message about fixing ?
also about change is unused you mean about monkey patching or require statement?

**Comment by SergeyKishenin** [comment]
No, just describe what was fixed in this — https://github.com/rharriso/bower-rails/blob/master/CHANGELOG.md — file

Yes, I mean Hash class extension.

**Comment by gingray** [comment]
@SergeyKishenin about hash, method except it rails method this part of code was not covered and thats why everything goes fine. it uses in this code
```
      if json.except('lib', 'vendor').empty?
        folders = json.keys
      else
        raise "Assuming a standard bower package but cannot find the required 'name' key" unless !!json['name']
        folders = ['vendor']
      end
```
https://github.com/rharriso/bower-rails/blob/master/lib/bower-rails/performer.rb#L74-L79
changelog updated

**Comment by SergeyKishenin** [comment]
Ah, I see, makes sense. But I think it is better to place this extension somewhere in the `lib` folder and require it if Hash does not respond to `:except`. How do you think?

**Comment by gingray** [comment]
@SergeyKishenin this method define in rails ActiveSupport::CoreExtensions it will be more sense if gem used outside rails.

**Comment by SergeyKishenin** [comment]
> this method define in rails ActiveSupport::CoreExtensions

I know. I mean that it looks hacky to define ext in `spec_helper`. For me, better solution is to define it in `lib` and require in spec_helper.

**Comment by gingray** [comment]
@SergeyKishenin i think its not good idea seems like 'npm kik situation', What you think about this solution?
```ruby
      if json.reject{ |key| ['lib', 'vendor'].include? key }.empty?
        folders = json.keys
      else
        raise "Assuming a standard bower package but cannot find the required 'name' key" unless !!json['name']
        folders = ['vendor']
      end
```

**Comment by SergeyKishenin** [comment]
@gingray ok, let's do like that

**Comment by gingray** [comment]
@SergeyKishenin add changes

**Comment by gingray** [comment]
@SergeyKishenin do you need some other changes ? Bcz seems pull request hanging in the air.

**Comment by SergeyKishenin** [comment]
@gingray Sorry for the delay. Merging

**Comment by inlikealion** [comment]
Has this been released? Because I'm still unable to run `rake bower:clean` with only a `bower.json` file.

**Comment by SergeyKishenin** [comment]
@inlikealion not yet, please, use the edge gem version

**Comment by gingray** [comment]
@inlikealion do you manually do rake bower:clean or you just get this error while usig rake bower:install ?

**Comment by inlikealion** [comment]
@gingray I run it manually.

**Pull request merged**